xlsx/src/export/mod.rs

```diff
@@ -119,14 +119,12 @@ pub fn save_xlsx_to_writer<W: Write + Seek>(model: &Model, writer: W) -> Result<
         let max_row = dimension.max_row;
         let sheet_dimension_str = &format!("{column_min_str}{min_row}:{column_max_str}{max_row}");
         let is_sheet_selected = selected_sheet as usize == sheet_index;
-        zip.write_all(
-            worksheets::get_worksheet_xml(
-                worksheet,
-                &model.parsed_formulas[sheet_index],
-                sheet_dimension_str,
-                is_sheet_selected,
-            )
-            .as_bytes(),
+        worksheets::write_worksheet_xml(
+            worksheet,
+            &model.parsed_formulas[sheet_index],
+            sheet_dimension_str,
+            is_sheet_selected,
+            &mut zip,
```

Comment:
I noticed that writing out the worksheet caused a memory spike of around 1 GB. That's because we were generating the whole worksheet as a String, then writing it in one go. Instead, pass in the writer and write it out incrementally to avoid the spike.